doc/source/development/rfc/rfc96_deferred_plugin_loading.rst

```diff
@@ -0,0 +1,499 @@
+.. _rfc-96:
+
+==================================================================
+RFC 96: Deferred C++ plugin loading
+==================================================================
+
+============== =============================================
+Author:        Even Rouault
+Contact:       even.rouault @ spatialys.com
+Started:       2023-Nov-01
+Status:        In development
+Target:        GDAL 3.9
+============== =============================================
+
+Summary
+-------
+
+This RFC adds a mechanism to defer the loading of C++ plugin drivers to
+the point where their executable code is actually needed, and converts a number
+of relevant drivers to use that mechanism. The aim is to allow for more modular
+GDAL builds, while improving the performance of plugin loading.
+
+It mostly targets in-tree plugin drivers, but it also provides a way for
+out-of-tree plugin drivers to benefit from deferred loading capabilities,
+provided libgdal is built in a specific way
+
+Context and motivation
+----------------------
+
+There are currently two ways of loading a GDAL C++ driver:
+
+- embedded in the core libgdal library. This is the default behavior
+  for drivers in the official GDAL source repository.
+
+- available in a shared library (.so, .dll, .dylib) in a directory where it
+  is dynamically loaded when GDALAllRegister() is called. This is what
+  out-of-tree drivers use, or in-tree drivers if enabling the
+  [GDAL|OGR]_ENABLE_DRIVER_FOO_PLUGIN=ON or GDAL_ENABLE_PLUGINS=ON
+  CMake options (cf https://gdal.org/development/building_from_source.html#build-drivers-as-plugins)
+  to build them as plugins.
+
+For packagers/distributors, the second option is convenient for in-tree drivers
+that depend on external libraries that are big and/or have a big number of
+dependencies (libparquet, etc) and that would substantially increase the size of
+the core libgdal library, or which have licenses more restrictive than the MIT
+license used by libgdal.
+
+However, there is a penalty at GDALAllRegister() time. For example, on Linux,
+it takes ~ 300 ms to complete for a build with 126 plugins, which is a substantial
+time for short lived GDAL-based processes (for example a script which would run
+gdalinfo or ogrinfo on many files). This time is entirely spent in the dlopen()
+method of the operating system and there is, to the best of our knowledge,
+nothing we can do reduce it... besides limiting the amount of dynamic loading
```

Comment:
Mising "to": "nothing we can do to reduce it"

Reply:
added